whois-nrtm4/src/main/java/net/ripe/db/nrtm4/DeltaFileGenerator.java

```diff
@@ -1,58 +1,50 @@
 package net.ripe.db.nrtm4;
 
-import net.ripe.db.nrtm4.persist.DeltaFileModelRepository;
+import net.ripe.db.nrtm4.persist.DeltaFileRepository;
 import net.ripe.db.nrtm4.persist.NrtmSource;
 import net.ripe.db.nrtm4.persist.NrtmVersionInfo;
 import net.ripe.db.nrtm4.persist.NrtmVersionInfoRepository;
 import net.ripe.db.nrtm4.publish.PublishableDeltaFile;
+import net.ripe.db.nrtm4.util.NrtmFileUtil;
 import net.ripe.db.whois.common.dao.SerialDao;
 import net.ripe.db.whois.common.domain.serials.SerialEntry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 import java.util.Optional;
 
 
 @Service
-public class NrtmProcessor {
+public class DeltaFileGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeltaFileGenerator.class);
 
-    private final DeltaFileModelRepository deltaFileModelRepository;
     private final DeltaTransformer deltaTransformer;
     private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
-    private final SnapshotSynchronizer snapshotSynchronizer;
     private final SerialDao serialDao;
+    private final DeltaFileRepository deltaFileRepository;
+    private final JsonSerializer jsonSerializer;
+    private final NrtmFileUtil nrtmFileUtil;
 
-    public NrtmProcessor(
-        final DeltaFileModelRepository deltaFileModelRepository,
+    public DeltaFileGenerator(
         final DeltaTransformer deltaTransformer,
         final NrtmVersionInfoRepository nrtmVersionInfoRepository,
-        final SnapshotSynchronizer snapshotSynchronizer,
-        final SerialDao serialDao
-        ) {
-        this.deltaFileModelRepository = deltaFileModelRepository;
+        final SerialDao serialDao,
+        final DeltaFileRepository deltaFileRepository,
+        final JsonSerializer jsonSerializer,
+        final NrtmFileUtil nrtmFileUtil
+    ) {
```

Comment:
You do not need to inject NrtmFileUtil as a component?

Reply:
done -- though since we're using Spring i'd normally prefer to make it injectable like pretty much every other functional class.

Reply:
We should not make util classes as injectable. Just because anything could be wired as a Spring bean doesn't mean everything should be wired as a Spring bean. Thanks for making this change. This keeps the code simple